### PR TITLE
🧹 remove the remaining wordpress pieces

### LINKED ISSUE #3565: Remove Wordpress from the dev setup
## Context

We have just eliminated Wordpress from our stack, but we have not yet removed it from our dev environment.

## Proposed improvement

Remove Wordpress db fetching and spin-up from the `make up.full` step.

If this makes `make up.full` basically empty, prune `make up.full`.

## Technical questions

- Do we want to preserve a different Makefile flow between the public and staff
  - E.g. should `.env.example-full` become `.env.example-staff` ?
- Is it clear what .env variables and what data you need for a full local dev experience for staff?

### PR DESCRIPTION
**Pull request by danyx23**
Fixes #3565

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,8 +33,6 @@
         "startTmuxServer": "node_modules/tmex/tmex dev \"yarn startTscServer\" \"yarn startLernaWatcher\" \"yarn startAdminServer\" \"yarn startViteServer\"",
         "startTscServer": "tsc -b -verbose -watch",
         "startViteServer": "vite dev",
-        "startWordpressPlugin": "cd ./wordpress/web/app/plugins/owid && yarn && yarn start",
-        "startSiteFront": "./devTools/vite/startVite.sh",
```
Comment (mlbrgl) [inline package.json:39]: this is still used by make up and make up.full. I'll revert it now to be able to continue testing, but feel free to amend if you had something else in mind.

File: docker-compose.full.yml [hunk 2]
```diff
@@ -79,42 +59,8 @@ services:
             - 8080:80
         volumes:
             - ./devTools/docker/vhosts.conf:/etc/nginx/conf.d/default.conf
-            - ./wordpress:/app
         depends_on:
             - fpm
```
Comment (mlbrgl) [inline docker-compose.full.yml:63]: `web` references the `fpm` container and `vhosts.conf`, which both don't exist anymore. `web` is not needed at this point though, so we should just remove it.

File: docs/setup-options-overview.md [hunk 3]
```diff
@@ -12,10 +12,6 @@ At a minimum, all of the options will transpile and serve the Grapher code, whic
 
 This needs a working MySQL database and gives you an interactive admin UI for visually editing our charts as well as all our ~4000 charts and their data.
 
-**Full WordPress setup** 🚀
```
Comment (mlbrgl) [inline docs/setup-options-overview.md:15]: if we're keeping `make up.full`, I think we should be coherent and explain here what this setup is for. However, t this point, the only difference I could notice between `make up` and `make up.full` was related to the cloudflare functions dev server. It used to be that the overhead of the `full` variant justified the split and associated maintenance, but this is much less true now, and we should just merge them and have a single local dev command.

**Review by mlbrgl — CHANGES_REQUESTED** [review]
Almost there! 👏

I've fixed a couple of regressions, and a few things still need a bit of attention, but this is overall looking pretty nice!

I still need to go over images, and will confirm when done.

On the discussion front, we should consider whether keeping `make up.full` still makes sense, given the small delta with `make up` (see comment).

File: adminSiteServer/mockSiteRouter.tsx [hunk 4]
```diff
@@ -379,16 +378,6 @@ mockSiteRouter.get("/headerMenu.json", async (req, res) => {
...
-mockSiteRouter.use(
-    // Not all /app/uploads paths are going through formatting
-    // and being rewritten as /uploads. E.g. blog index images paths
-    // on front page.
-    ["/uploads", "/app/uploads"],
-    express.static(path.join(WORDPRESS_DIR, "web/app/uploads"), {
-        fallthrough: false,
-    })
-)
-
```
Comment (mlbrgl) [inline adminSiteServer/mockSiteRouter.tsx:391]: images on the remaining wordpress pages will not render now that they are not available locally, so we should probably redirect to R2 (or document it so we know it is a wontfix)

**Review by mlbrgl — COMMENTED** [review]
my understanding of what's left of the [wordpress images baking flow](https://github.com/owid/ops/blob/57b50b942abcb6959bf15b606bf8fae7410720b5/templates/owid-cloudflare-prod/README.md#wordpress-images-baking-flow):

Screenshot 2024-05-02 at 21 40 30

The master copy of wordpress images is now R2, and updates to Wordpress uploads (/uploads) [should be made directly on R2](https://www.notion.so/owid/Editing-old-wordpress-posts-e-g-about-pages-6ea77093df0344b1a6c1bf5cfe47c51b?pvs=4#76ab5a35a5b9429b8dad38778c97797f).

I've finished my review, let me know if you want to discuss certain points.

File: package.json (hunk 1 above)
Comment (danyx23) [reply package.json:39]: Thanks, that was a mistake :)

File: Makefile [hunk 5]
```diff
@@ -137,8 +134,6 @@ migrate:
 	@echo '==> Running DB migrations'
 	rm -rf itsJustJavascript && yarn && yarn buildLerna && yarn buildTsc && yarn runDbMigrations
 
-refresh.full: refresh refresh.wp sync-images
-
 refresh:
```
[the remark this answers is not in the document]
Comment (danyx23) [reply Makefile:137]: Nice idea - I added that

File: adminSiteClient/PostsIndexPage.tsx [hunk 6]
```diff
@@ -222,14 +222,6 @@ class PostRow extends React.Component<PostRowProps> {
                 <td>
                     <Timeago time={post.updatedAtInWordpress} />
```
[the remark this answers is not in the document]
Comment (danyx23) [reply adminSiteClient/PostsIndexPage.tsx:223]: Hilarious, this has been broken for a long time :)

File: baker/startDeployQueueServer.ts [hunk 7]
```diff
@@ -10,6 +10,7 @@ import * as db from "../db/db.js"
 // by registering listeners on SIGINT.
 import "../db/cleanup.js"
 
+// TODO: I think we can delete this script but I'm not entirely sure.
```
[the remark this answers is not in the document]
Comment (danyx23) [reply baker/startDeployQueueServer.ts:13]: Right - thanks for clarifying! I added a comment to communicate why we still have this.

File: adminSiteServer/mockSiteRouter.tsx (hunk 4 above)
Comment (danyx23) [reply adminSiteServer/mockSiteRouter.tsx:391]: Good catch - I added a new route that redirects to assets.ourworldindata.org for wp images

File: ops/buildkite/deploy-content [hunk 8]
```diff
@@ -55,16 +54,6 @@ update_owid_content_repo() {
     )
 }
 
-sync_wordpress_uploads() {
```
[the remark this answers is not in the document]
Comment (danyx23) [reply ops/buildkite/deploy-content:58]: Removed it

**Comment by danyx23** [comment]
@mlbrgl thanks a lot for your thorough review, much appreciated! I have:
- added a `refresh.full` command
- removed the docker-compose.full.yaml file
- kept the "make up.full" command even if that only differs in the CF worker setup - but if we merge this into "make.up" then when you run it you are asked to authenticate to CF in a browser and that will confuse non-staff people running our setup
- added a redirect in mocksite router for old images so that images are fetched by the client from assets.ourworldindata.org
- removed some code you mentioned was redundant
- fixed the updated_at_in_wordpress bug :)
- and rebased on top of master

It would be great if you could have one last look at the last commit, then I think we could ship it!

**Review by mlbrgl — APPROVED** [review]
that all makes sense, thanks for update!

I've done a couple more cleanup tasks, and rebaked the whole site on staging. Everything looks good.

**Pull request merged**